pkg/common/plugin/k8s/apiserver/client_test.go

```diff
@@ -1,23 +1,154 @@
+MIIDADCCAeigAwIBAgIBAjANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQDEwptaW5p
+a3ViZUNBMB4XDTE5MDQzMDEyNDkxNloXDTIwMDQzMDEyNDkxNlowMTEXMBUGA1UE
+ChMOc3lzdGVtOm1hc3RlcnMxFjAUBgNVBAMTDW1pbmlrdWJlLXVzZXIwggEiMA0G
+CSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDIMCT2+uEDVfz8BNUoE5wa4mAqr8QL
+kJknjUFS3kdRke/SIiuPgAi1GaQK4XkT24fTIqYO8YwlwSYcsymJ0E/KHzns+kfn
+OS4ls0fXibkB2lw36q6VltNkBGs7fwD0De6DP+PP/89eTSStXaqfz5lmpbjdsUM6
+P8zeMgkJoPNdf1bYikdRwAVuhhdW1pFbHNVdqQMCVFYwhWrav5r8RBHERR7aUwx4
+T3RMPtN9yb6OPLVrycUKHEi8N5J4aYwczu2QZ/AUdriapB4QrdL1ePkBI0q0LOww
+2RDbfPKd/Y5N9FFbTAkJie9TaiffhxW9FTYz/OJlhKBALH9InKYoatNLAgMBAAGj
+PzA9MA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUH
+AwIwDAYDVR0TAQH/BAIwADANBgkqhkiG9w0BAQsFAAOCAQEAQ4S/JRl7IqKHCEuN
+TEhEGAs7A2DJLrKM1zVUeP4LHSXdQG9px0oaGvONAgKjtY+3j1ajecCPKHeduVJ2
+RFOqlR0jx74vnwat/C9ELAlFAyvwRzVMxoF1a3SuAq1D62MU3smD03X3WOlUrgpU
+Ispvk1GICnSys++AacjyNTKlRUUheDdSObHQpYt7MOl1nygHl9HpGWxvTaXCiz2y
+RZUI/exII+oNBrwRv2b3Hmflm5sG93siVSvZ0EXI27O3NjvJBPryKyJ/9A6uq975
+G8cDWzZ5QYzlKr1qcuYaP5Aw7DbMVIU17vVACili6R9WD9+wk2rjSmS737YJ+Ud9
+vOjlQw==
+-----END CERTIFICATE-----`)
+
+	kubeConfigClientKey = []byte(`-----BEGIN RSA PRIVATE KEY-----
+MIIEpAIBAAKCAQEAyDAk9vrhA1X8/ATVKBOcGuJgKq/EC5CZJ41BUt5HUZHv0iIr
+j4AItRmkCuF5E9uH0yKmDvGMJcEmHLMpidBPyh857PpH5zkuJbNH14m5AdpcN+qu
+lZbTZARrO38A9A3ugz/jz//PXk0krV2qn8+ZZqW43bFDOj/M3jIJCaDzXX9W2IpH
+UcAFboYXVtaRWxzVXakDAlRWMIVq2r+a/EQRxEUe2lMMeE90TD7Tfcm+jjy1a8nF
+ChxIvDeSeGmMHM7tkGfwFHa4mqQeEK3S9Xj5ASNKtCzsMNkQ23zynf2OTfRRW0wJ
+CYnvU2on34cVvRU2M/ziZYSgQCx/SJymKGrTSwIDAQABAoIBAQCQdp3WVcmXnLAK
+NnqUh0I57G81ns29AsZjUn53jMyuwr/sRphk4CJofm5xI7E6cUwdQ33OfuWCQVZ9
+k5VATMGnvM0ShLLq28q/jhckJdEK00eFWqhObx9xp/ayYr6PYJZkxPBjo9lD1ivH
+qDZ/SVMMTj+QTGGVYYE4P6dh+XJmXzvIHVJmG2ZClSlHeNAze+WygkOykZnQF1do
+JvqBxl5YHUM8PSJ2xnMYpHVFJkmAp0GntpgxgxR6yBPLroQSeU/SJUQBhinstQ++
+v9P3E5eq4VTDGjWZYzAg95boLGowhsQupNHMmc3TqqJkXpGPaXzFe/O5ZiVzrJ8y
+UlVye7dxAoGBAMpyqXE4ctfN1hNRqFwsj23MKZMekOUk9vXZiWkguIep48zp8PSy
+hThl+h37ddk5jqRA8aiVeMXkvaIzs2b/8wezuu+4xtCdPEdZu/xAw4x0yJ5Uv0E1
+Ci6y0bBmcV0zv4fRsNvErMAkUU8bo2oPbNT9siRRfi/HEQTjC5gnCwZtAoGBAP0k
+c46TQZ07YccWskDDkf/KAlqBCPlo6ar/CJp0Djqjap9/mNMl0lJV7dpy/WS0BP9S
+LyQb5FYsv7ga3iMdi1oEGQsEc31nePDyYZRp/aD0MNgQUXv60Zevyd05NSkTRb9i
+0ob5vALuOczqjeL3YF1oLH0HZyG0bs+oCkjL4I2XAoGAMcodbiEJ7ZVMDxhIJdM3
+uzM5Dlu4GwMKUdCcgORxPbxEsELg5e154jMCXplXlIMZV8A5LtMEDveAxAGfH7fX
+F4/Wa9qv2uKwzoN9Pj7XWRXnuTjyiKD4zh9gftfTDa8Kbebebk5ihibocGJFwHHm
+vENgqpn4RNvaja8hTNxdU8kCgYBTKlWYosJswKSX/tnjMx1VNu3dBAWJwzD5j74o
+2DYQh72w1v/DZuqZSEfTe/HJ0ubNZxe7ujojIaJ+/ry6NqugkDYWC4lRytvN9SOf
+2c6MwY0Gfx32KGoRdpxQRMo1S3KftPzLgWKGZ/OvYePpjDIpnd732KXGSfwZ1vBC
+CFEm0wKBgQCwARG9qV4sUoBvwLyBHQbPFZi/9PYwvDsnzjmKTUPa+kd4ATrv7gBY
+oN1CqmWqJQYVB6oGxFMaebeijY82beDN3WSBAK2FGvmdi3vZUAHHXyNOBS2Wq6PA
+oIrPuyjOmscrC627wX3LGUHwPKtNArBT8lKFfda1B1BqAk0q1/ui/A==
+-----END RSA PRIVATE KEY-----`)
 )
 
 func TestAPIServerClient(t *testing.T) {
 	spiretest.Run(t, new(ClientSuite))
 }
 
+// Mocks for this test suite were generated with mockgen follows:
```

Comment:
We should provide a `generate.go` that generates the mocks via `go generate` for each of these mocks (see other mock packages under `test/mock`).